proxy/tests/discovery.rs

```diff
@@ -18,6 +18,21 @@ fn outbound_asks_controller_api() {
     assert_eq!(client.get("/bye"), "bye");
 }
 
+#[test]
+fn outbound_asks_controller_about_authority() {
```

Comment:
Please factor out the common logic in these tests so that they are easier to review. These tests all look very similar, but there are at least trivial differences, and it isn't clear if the differences are intentional. It's too hard to tell the difference in what each test is testing now.

Some tests have the comment "don't set srv as outbound(), so that SO_ORIGINAL_DST isn't" but this one doesn't. Also this test is formatted differently than the others, which makes things more confusing. If we factor out the commonality we can prevent this.

Reply:
@briansmith the latest state of #556 should improve this significantly --- i can merge that back onto this branch if you like.